src/RBACHandler.ts

```diff
@@ -19,20 +28,53 @@ export class RBACHandler implements Authorization {
 
     private readonly rules: RBACConfig;
 
-    constructor(rules: RBACConfig) {
+    private readonly fhirVersion: FhirVersion;
+
+    constructor(rules: RBACConfig, fhirVersion: FhirVersion) {
         this.rules = rules;
         if (this.rules.version !== this.version) {
             throw Error('Configuration version does not match handler version');
         }
+        this.fhirVersion = fhirVersion;
     }
 
     isAuthorized(request: AuthorizationRequest): boolean {
         const decoded = decode(request.accessToken, { json: true }) || {};
         const groups: string[] = decoded['cognito:groups'] || [];
 
+        if (request.exportType) {
+            return this.isExportAllowed(groups, request.exportType);
+        }
         return this.isAllowed(groups, request.operation, request.resourceType);
     }
 
+    private isExportAllowed(groups: string[], exportType: ExportType): boolean {
+        for (let index = 0; index < groups.length; index += 1) {
+            const group: string = groups[index];
+            if (this.rules.groupRules[group]) {
+                const rule: Rule = this.rules.groupRules[group];
+                if (exportType) {
+                    if (exportType === 'system') {
+                        if (
+                            (this.fhirVersion === '4.0.1' &&
+                                isEqual(rule.resources.sort(), SUPPORTED_R4_RESOURCES.sort())) ||

```

Comment:
What happens if a group is only allowed to `create` all the resource types? Are we missing a check to see it the operation `read` is present?

Reply:
Good catch. Fixing.